### DISCUSSION TITLE
Can Docker action receive a list as an input argument?

### DISCUSSION THREAD

**Discussion by dingo-d** [community/community#26938] [Actions]
<p>When creating my own Docker action, I can pass arguments to it using <code>with</code> key in my workflow yml file. For instance</p>
<pre><code>- name: Run tests in the container
              uses: dingo-d/plugin-test-action@v1.0.23
              with:
                wp-db-name: 'wordpresstest'
                wp-db-user: 'admin'
                wp-db-pass: 'password'
                wp-db-host: 'mysql'
                wp-user: 'admin'
                wp-pass: 'password'
</code></pre>
<p>In my action’s entrypoint.sh script I can then reference the <code>wp-db-name</code> with $1, <code>wp-db-user</code> with $2, etc.</p>
<p>Is it possible to have a key that will have a value that is an array, like:</p>
<pre><code>option: [option1, option2, option3]
</code></pre>
<p>If that is possible, how would I reference that in my entrypoint.sh?</p>
<p>Are those inputs passed as env variables inside the script that I could then reference with names instead of numbers, like <code>$wp-db-name</code> etc.?</p>
<p>The aim is to reference that key with array value so that I can loop through it in my bash script.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/dingo-d">@dingo-d</a> , </p>
<p>The input variable type only supports string , number and boolean. So an input key with array value is not supported. You could consider to convert the string into array in entrypoint.sh . </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7e91cdf242e94b2f4445fdad206908f39ea03902.png" data-download-href="/uploads/short-url/i3GxQykK1o7Vbt5suqMRsuSyPWq.png?dl=1" title="input value.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181100880-390deee8-a1c3-49b2-9acb-6c8b9e303f93.png" alt="input value.png" data-base62-sha1="i3GxQykK1o7Vbt5suqMRsuSyPWq" width="690" height="245" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7e91cdf242e94b2f4445fdad206908f39ea03902_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">input value.png</span><span class="informations">794×282 6.13 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><em>&gt;&gt;Are those inputs passed as env variables inside the script that I could then reference with names instead of numbers, like <code>$wp-db-name</code> etc.? </em></p>
<p>There are env variables for inputs when using the docker container action. But the env name has an INPUT_  prefix and all the letters are in uppercase.  Please note that ‘-’ is an invalid charture in shell variable name, you could use ‘_’ instead. I change all the input variables to use ‘_’ .  Such as ‘wp_db_name’ .</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5f3446238ded35e0cdded75ccba9119a21d4d72a.png" data-download-href="/uploads/short-url/dAdm59gB2CYweENjP6dUzMm5XlM.png?dl=1" title="input env.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181100881-ead710cd-ca32-4b1f-8d4c-f621529a5f8c.png" alt="input env.png" data-base62-sha1="dAdm59gB2CYweENjP6dUzMm5XlM" width="690" height="347" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5f3446238ded35e0cdded75ccba9119a21d4d72a_2_690x347.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/5f3446238ded35e0cdded75ccba9119a21d4d72a_2_1035x520.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/5f3446238ded35e0cdded75ccba9119a21d4d72a_2_1380x694.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5f3446238ded35e0cdded75ccba9119a21d4d72a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">input env.png</span><span class="informations">1486×748 38.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You could use $INPUT_WP_DB_USER in your entrypoint.sh. </p>
<p>There is the action setting in workflow yml: </p>
<pre><code>- uses: ./
      with:
        who_to_greet: '[1,2,3]'
        wp_db_name: 'wordpresstest'
        wp_db_user: 'admin'
</code></pre>
<p>And entrypoint.sh example:</p>
<pre><code>#!/bin/sh -l

echo "Hello $1"
time=$(date)
echo $INPUT_WHO_TO_GREET
echo $INPUT_WP_DB_USER
echo ::set-output name=time::$time

newstr=$(echo $INPUT_WHO_TO_GREET | sed 's/\[//g' ) # remove [
newstr2=$(echo $newstr | sed 's/\]//g' ) # remove ]
echo $newstr2
IFS=, # split on ,
set -o noglob # disable glob
set -- $newstr2 # split+glob with glob disabled.
echo "$# elements:"
printf '"%s"\n' $@
</code></pre>
<p>  </p>

**Comment by dingo-d** [comment]
<p>Thank you for the explanation! I’ll see about using string and converting it to array in the entrypoint, that’s a useful tip, as well as with the inputs :) </p>
